### PR TITLE
Update deployment flow state

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6190/
Test FAILed.

**Comment by anthonydillon** [comment]
@huwshimi Nice to see the new state stuff. Not sure if this is ready for review as there is no QA steps but code looks good. Few tests failing though.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1774,7 +1784,13 @@ YUI.add('juju-gui', function(Y) {
           this._renderMachineView.bind(this),
           this._clearMachineView.bind(this)],
         ['gui.inspector',
-          this._renderInspector.bind(this)]
+          this._renderInspector.bind(this)],
+        ['gui.deploy',
+          this._renderDeployment.bind(this),
+          this._clearDeployment.bind(this)],
+        ['deploy',
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: We shoudln't have a base 'deploy' state required here - I also don't see where you've changed any of the changeState's to add a non `gui` state 'deploy'. Was this left in?

Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: I have no idea why, but I was getting messages for both "No dispatcher found for key: deploy" and "No dispatcher found for key: gui.deploy".

**Pull request merged**